### PR TITLE
Feature workout

### PR DESCRIPTION
**Pull request by KevinTss**

### CODE REVIEW

File: src/_utils/constants/index.js [hunk 1]
```diff
@@ -38,23 +40,76 @@ const TYPE = {
 }
 
 const UNIT = {
-  HEIGHT: {
+  DISTANCE: {
     CM: 'cm',
     IN: 'in',
```
Comment (alvermeylen) [inline src/_utils/constants/index.js:45]: Shouldn't we split height and distance? cm and in will probably only be used for height while the others for distance.

File: src/controllers/coach/index.js [hunk 2]
```diff
@@ -189,6 +190,33 @@ const searchUserAsCoach = async (req, res) => {
...
+
+    const language = await getLangByISO(lang)
+    const workout = await createWorkout(
+      user._id,
+      language._id.toString(),
+      title,
+      content,
+    )
+
+    res.status(201).json(workout)
```
Comment (alvermeylen) [inline src/controllers/coach/index.js]: Why 201 instead of  200 ?

File: src/controllers/coach/index.js [hunk 3]
```diff
@@ -189,6 +190,33 @@ const searchUserAsCoach = async (req, res) => {
...
+  } catch (error) {
+    res.status(500).json({
+      public_message: 'Error in adding customer to coach',
```
Comment (alvermeylen) [inline src/controllers/coach/index.js]: copy paste :p

File: src/controllers/exercise/index.js [hunk 4]
```diff
@@ -54,12 +54,7 @@ module.exports = {
 
       if (!userOwnerId) throw new Error('userOwnerId is required')
 
-      const acceptedLanguagesValue = Object.keys(LANG).map((k) =>
-        LANG[k].NAME.toLowerCase(),
-      )
-
-      if (!acceptedLanguagesValue.includes(lang))
-        throw new Error('Lang is invalid')
+      if (!ACCEPTED_LANGS.includes(lang)) throw new Error('Lang is invalid')
```
Comment (alvermeylen) [inline src/controllers/exercise/index.js:57]: Isn't that business logic that should be managed by the handler ?

File: src/controllers/exercise/index.js [hunk 5]
```diff
@@ -54,12 +54,7 @@ module.exports = {
...
 
       const language = await getLangByISO(lang)
       const userOwner = await getUserById(userOwnerId)
```
Comment (alvermeylen) [inline src/controllers/exercise/index.js:60]: not needed, just use the id from params

File: src/controllers/workout/handlers.js [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+    if (!langId) throw new Error('langId is required')
+
+    if (!ObjectId.isValid(langId)) throw new Error('langId is incorrect')
+
+    if (!title) throw new Error('title is required')
+
+    const newWorkout = await createNewWorkout(
+      userOwnerId,
+      langId.toString(),
```
Comment (alvermeylen) [inline src/controllers/workout/handlers.js]: langId is already a String, no?

File: src/controllers/workout/handlers.js [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    if (!ObjectId.isValid(id)) throw new Error('workout id is incorrect')
+
+    return await updateWorkoutById(id, workoutData)
+  },
+
+  /**
+   * @param {string} id required
+   */
+  removeWorkout: async (id) => {
+    if (!id) throw new Error('WOrkout id is required')
```
Comment (alvermeylen) [inline src/controllers/workout/handlers.js]: typo: "workout" not "WOrkout"

File: src/controllers/workout/index.js [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+      if (!lang) throw new Error('Lang is required')
+
+      // if (!ownerId) throw new Error('ownerId is required')
+
+      if (!ACCEPTED_LANGS.includes(lang)) throw new Error('Lang is invalid')
+
+      const language = await getLangByISO(lang)
+      const newWorkout = await createWorkout(
+        user._id,
+        language._id.toString(),
```
Comment (alvermeylen) [inline src/controllers/workout/index.js]: do we need the .toString() or not for _id fields ?

File: src/models/workout.js [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+  isPrivate: {
+    type: Boolean,
+    default: false,
+  },
+
+  /**
+   * Translatable content:
+   * The content who can be written into different languages
+   */
+  content: [workoutContentSchema],
```
Comment (alvermeylen) [inline src/models/workout.js:34]: WHere are the exercises ?

File: src/_utils/constants/index.js (hunk 1 above)
Comment (KevinTss) [reply src/_utils/constants/index.js:45]: As discussed probably better to use `length`

File: src/controllers/coach/index.js (hunk 2 above)
Comment (KevinTss) [reply src/controllers/coach/index.js]: https://restfulapi.net/http-status-codes/

**Pull request merged**